src/modules/navigator/mission_block.cpp

```diff
@@ -711,7 +711,13 @@ MissionBlock::set_land_item(struct mission_item_s *item, bool at_current_locatio
 	if (at_current_location) {
 		item->lat = _navigator->get_global_position()->lat;
 		item->lon = _navigator->get_global_position()->lon;
-		item->yaw = _navigator->get_global_position()->yaw;
+
+		if (_navigator->home_position_valid()) {
+			item->yaw = _navigator->get_global_position()->yaw;
+
+		}	else {
+			item->yaw = _navigator->get_local_position()->yaw;
+		}
```

Comment:
We can just always use `_navigator->get_local_position()->yaw;` and get rid of the special case.